object/slicer/slicer_test.go

```diff
@@ -269,6 +269,23 @@ func testSlicer(t *testing.T, size, sizeLimit uint64) {
 	}
 }
 
+// eofOnLastChunkReader is a special reader for tests. It returns io.EOF with the last data portion.
+type eofOnLastChunkReader struct {
+	Payload []byte
+	point   int
+}
+
+func (l *eofOnLastChunkReader) Read(p []byte) (int, error) {
```

Comment:
lets also return `0, nil` to test all `io.Reader` doc statements? For example 
```go
func (l *eofOnLastChunkReader) Read(p []byte) (int, error) {
  if !l.readEmpty {
    l.readEmpty = true
    return 0, nil
  }
```

Reply:
Extended with new test

Reply:
does it ever return `0, nil`?

Reply:
No, depending on the condition

Reply:
lets add then, it's simple (see initial comment) and tests possible `io.Reader` behavior

Reply:
Updated